bindings/go/runtime/registry.go

```diff
@@ -0,0 +1,221 @@
+	return func(registry *Scheme) {
+		registry.allowUnknown = allowUnknown
+	}
+}
+
+func (r *Scheme) Clone() *Scheme {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	clone := NewScheme(WithAllowUnknown(r.allowUnknown))
+	maps.Copy(r.types, clone.types)
+	return clone
+}
+
+func (r *Scheme) RegisterWithAlias(prototype any, types ...Type) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	for _, typ := range types {
+		if _, exists := r.types[typ]; exists {
+			return fmt.Errorf("type %q is already registered", typ)
+		}
+		r.types[typ] = prototype
+	}
+	return nil
+}
+
+// GetTypeFromAny uses reflection to extract the "Type" field from any struct.
+func GetTypeFromAny(v any) (Type, error) {
+	val := reflect.ValueOf(v)
+
+	// Ensure v is a struct or a pointer to a struct
+	if val.Kind() == reflect.Ptr {
+		val = val.Elem()
+	}
+
+	if val.Kind() != reflect.Struct {
+		return Type{}, fmt.Errorf("expected struct, got %s", val.Kind())
+	}
+
+	// Get the field by name
+	field := val.FieldByName("Type")
+	if !field.IsValid() {
+		return Type{}, fmt.Errorf("field 'Type' not found")
+	}
+
+	// Ensure it's of Type type
+	if field.Type() != reflect.TypeOf(Type{}) {
+		return Type{}, fmt.Errorf("field 'Type' is not of expected Type struct")
+	}
+
+	// Return the Type value
+	return field.Interface().(Type), nil
+}
+
+func (r *Scheme) MustRegister(prototype any, version string) {
+	t := reflect.TypeOf(prototype)
+	if t.Kind() != reflect.Pointer {
+		panic("All types must be pointers to structs.")
+	}
+	t = t.Elem()
+	r.MustRegisterWithAlias(prototype, NewUngroupedVersionedType(t.Name(), version))
+}
+
+func (r *Scheme) TypeForPrototype(prototype any) (Type, error) {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+
+	for typ, proto := range r.types {
+		// if there is an unversioned type registered, do not use it
+		// TODO find a way to avoid this or to fallback to the fully qualified type instead of unqualified ones
+		if !typ.HasVersion() {
+			continue
+		}
+		if reflect.TypeOf(prototype).Elem() == reflect.TypeOf(proto).Elem() {
+			return typ, nil
+		}
+	}
+
+	return Type{}, fmt.Errorf("prototype not found in registry")
+}
+
+func (r *Scheme) MustTypeForPrototype(prototype Typed) Type {
+	typ, err := r.TypeForPrototype(prototype)
+	if err != nil {
+		panic(err)
+	}
+	return typ
+}
+
+func (r *Scheme) IsRegistered(typ Type) bool {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+
+	_, exists := r.types[typ]
+	return exists
+}
+
+func (r *Scheme) MustRegisterWithAlias(prototype any, types ...Type) {
+	if err := r.RegisterWithAlias(prototype, types...); err != nil {
+		panic(err)
+	}
+}
+
+// NewObject creates a new instance of types.Typed.
+func (r *Scheme) NewObject(typ Type) (Typed, error) {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+
+	var object any
+	// construct by full type
+	proto, exists := r.types[typ]
+	if exists {
+		t := reflect.TypeOf(proto)
+		for t.Kind() == reflect.Ptr {
+			t = t.Elem()
+		}
+		object = reflect.New(t).Interface()
+
+		return object.(Typed), nil
```

Comment:
should we type check here? Or is this just always true... ?

Reply:
I think its a bug. It will return any now because the Typed interface is not used anymore as binding (I changed to any and reflect)

Reply:
removed type assertion